pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -479,6 +494,19 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
+		// Ensure ownerRefs are set correctly for the condemned pods.
+		for i := range condemned {
```

Comment:
For the scaledown case, will this be able to handle pods that were manually deleted by a user?

Reply:
If the pod has been manually deleted, then yes, there is a race where the PVC won't get marked for deletion because there's no object to make an ownerRef to. I think this is similar to the case where the PVC delete policy is changed from retain to onScaledown during a scaledown.

The method used here retains conservative behavior, where if something unexpected is happening, PVCs won't be deleted. I think that makes sense: if a user is manually deleting pods, then manually deleting PVCs shouldn't be a burden. It would be worse I think to be more proactive about deleting and expose ourselves to races where we delete things that don't want to be deleted.